core/templates/pages/exploration-player-page/learner-experience/tutor-card.component.ts

```diff
@@ -271,6 +272,39 @@ export class TutorCardComponent {
     return this.translateService.instant(milestoneMessageTranslationKey);
   }
 
+  setNextMilestoneAndCheckIfProgressBarIsShown(): boolean {
+    if (
+      !this.inStoryMode ||
+      this.completedChaptersCount > 50 ||
+      (this.milestoneMessageIsToBeDisplayed &&
+        MILESTONE_SPECIFIC_COMPLETED_CHAPTER_COUNTS.includes(
+          this.completedChaptersCount))
+    ) {
+      this.nextMilestoneChapterCount = null;
+      return false;
+    }
+
+    if (
+      !this.milestoneMessageIsToBeDisplayed &&
+      MILESTONE_SPECIFIC_COMPLETED_CHAPTER_COUNTS.includes(
+        this.completedChaptersCount)
+    ) {
+      let chapterCountIndex = (
+        MILESTONE_SPECIFIC_COMPLETED_CHAPTER_COUNTS.indexOf(
+          this.completedChaptersCount));
+      this.nextMilestoneChapterCount = (
+        MILESTONE_SPECIFIC_COMPLETED_CHAPTER_COUNTS[chapterCountIndex + 1]);
+      return true;
+    }
+
+    for (let milestoneCount of MILESTONE_SPECIFIC_COMPLETED_CHAPTER_COUNTS) {
+      if (milestoneCount > this.completedChaptersCount) {
+        this.nextMilestoneChapterCount = milestoneCount;
+        return true;
+      }
+    }
```

Comment:
There is a missing return at the end of this function.

Reply:
Actually, based on all of the prior `if` conditions in the function, control will never reach the end of function, past the `for` loop. How do I handle this, i.e does there still need to be a return at the end?

Reply:
Moreover, I won't be able to write a Frontend test for it too (because control will never reach that part of the function no matter the combination of conditions), unless I change the function itself.

Reply:
I guess return false, in the tests you can mock `MILESTONE_SPECIFIC_COMPLETED_CHAPTER_COUNTS` so that the loop does finish.

Reply:
Got it

Reply:
Done.